rfcs/039-requesting-api-design/README.md

```diff
@@ -0,0 +1,215 @@
+# Requesting API design
+
+## Background
+
+As part of the work to add requesting, we need to add new some new APIs.
+In particular, these APIs should allow a user to:
+
+-   Get the status of a given item (e.g. available, on hold)
+-   Request an item
+-   See a list of their current requests
+-   Relinquish a pending request
+
+This RFC describes how these APIs will work, and provides examples of requests and responses.
+
+It does *not* suggest implementation details – those can be discussed separately.
+
+
+
+## Guiding principles
+
+1.  **We should not assume requests are sent to a particular service.**
+
+    Although the initial implementation will be making requests for items in Sierra, we may add support for requesting from other systems in a future update.
+    We should not assume the use of Sierra.
+
+2.  **We should use platform identifiers throughout.**
+
+    We may include source identifiers, but interactions with these APIs should start with platform identifiers (e.g. `a222wwjt`).
+    This is a consequence of principle (1).
+
+
+
+## Proposed APIs
+
+### Get the status of a given item
+
+We will add a new endpoint `/catalogue/v2/works/{workId}/items`.
+
+-   This will return a modified version of our existing `Work` model.
+
+    In particular, responses will have only two fields: `id` and `items`.
+    The `Item` model will have a new field (`status`) which returns an instance of a new `ItemStatus` type, e.g.
+
+    ```
+    "status": {
+      "id" : "available",
+      "label" : "Available",
+      "type": "ItemStatus"
+    }
+    ```
+
+-   This will return an item for every item in the corresponding `/works` request.
+
+    If an item cannot be requested (e.g. it is tracked in Calm, not Sierra), we will return a `not-requestable` status.
+    This includes items that only have digital locations.
+
+-   Using this endpoint will require an API key.
+
+    Requests to this endpoint will trigger requests to Sierra (and possibly other systems in future).
+    We don't want it to be used to DDOS those systems, so we need a way to turn off misbehaving clients.
+
+<details>
+<summary>Example request/response</summary>
```

Comment:
I think the front-end will continue to behave as it currently does – it has to get enough of the items to know it's worth making this request (no point requesting items on a sans-item work). The items are a useful thing it can render immediately, without hitting this API – in particular, digitised items.

If the initial request to /works succeeds but this API fails, we should still show the items on the page.